src/libs/error.mock.ts

```diff
@@ -0,0 +1,26 @@
+import _ from 'lodash/fp'
+
+/**
+ * Allows observation of when an error occurs via `errorWatcher`, i.e.:
+ *     expect(errorWatcher).toHaveBeenCalledTimes(1)
+ *     expect(errorWatcher).toHaveBeenCalledWith('my error message', expect.anything())
+ * The errorWatcher call counts and information will be cleared in between tests,
+ * thanks to Jest being configured to auto-clear mock between tests at the project level.
+ */
+export const errorWatcher = jest.fn()
+
+/**
+ * Provides a mocked version of error module's withErrorReportingInModal
+ */
+export const mockWithErrorReportingInModal = _.curry((title, onDismiss, fn) => {
+  const errorHandler = async () => {
+    try {
+      return await fn()
+    } catch (error) {
+      errorWatcher(title, error)
```

Comment:
`withErrorReportingInModal` rethrows the error after reporting it and dismissing the modal.

https://github.com/DataBiosphere/terra-ui/blob/2751494cc4ca3401e1a9d253a0ab2c5fd3cf1fc0/src/libs/error.js#L45-L51

Since this mocked version swallows errors, it could allow code that doesn't handle the re-thrown error to pass tests.

Reply:
Instead of creating this shared mock, would it be simpler to mock `reportError` in AnalysisDuplicator.test.ts?

Reply:
a minor fix to have the onDismiss happen in same spot in shared mock as with real implementation was agreed to.